src/libsyntax_pos/lib.rs

```diff
@@ -83,16 +83,49 @@ pub struct MultiSpan {
 }
 
 impl Span {
+    #[inline]
+    pub fn new(lo: BytePos, hi: BytePos, ctxt: SyntaxContext) -> Self {
+        Span { lo, hi, ctxt }
+    }
+
+    #[inline]
+    pub fn lo(self) -> BytePos {
+        self.lo
+    }
+    #[inline]
+    #[must_use]
+    pub fn set_lo(self, lo: BytePos) -> Span {
```

Comment:
I'm not sure `set_x` are the best possible names since the functions are not mutating, but at least I guarded them from mistakes by `#[must_use]`.